designer/controller.js

```diff
@@ -0,0 +1,173 @@
+const { nanoid } = require("nanoid");
+const utility = require('../utility/core');
+const jdlConverter = require('../utility/jsonToJdl');
+const exec = require("child_process").exec;
+const blueprintDao = require('../dao/blueprintDao');
+
+exports.test = function (req, res) {
+    return res.status(200).send({ message: "Test API for wda server" });
+};
+
+
+/**
+ * Get specific blueprint with given project Id
+ * @param {*} req 
+ * @param {*} res 
+ */
+exports.getBlueprint = function (req, res) {
+  blueprintDao.getByProjectId({project_id: req.params.project_id})
+  .then(result => {
+    console.log("Retrieved blueprint:", result);
+    if (Array.isArray(result) && result.length === 1) {
+        var uniqueResult = result[0];
+        return res.status(200).send(uniqueResult);
+      } else {
+        return res.status(200).send({result});
+      }
+  })
+  .catch(error => {
+    console.error("Error retrieving blueprint:", error);
+    return res.status(500).send({ message: "Error retrieving blueprint" });
+  });
+};
+
+/**
+ * Get all blueprints with given user Id
+ * @param {*} req 
+ * @param {*} res 
+ */
+exports.getBlueprints = function (req, res) {
+    blueprintDao.getByUserId({user_id: req.params.user_id})
+    .then(results => {
+      console.log("Retrieved blueprints:", results);
+      return res.status(200).send(results);
+    })
+    .catch(error => {
+      console.error("Error retrieving blueprints:", error);
+      return res.status(500).send({ message: "Error retrieving blueprints" });
+    });
+  };
+
+/**
+ * generates services/infrastructure code from blueprint
+ * @param {*} req 
+ * @param {*} res 
+ */
+exports.generate = function (req, res) {
+    const body = req.body;
+    // console.log(body)
+
+    console.log(
+        "Hi",
+        "Your project Name is",
+        body.projectName
+    );
```

Comment:
logs can be standardised, `console.log("Generating project: " + body.projectName + ", for user: " + userId)`